src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -608,141 +621,174 @@ object P
-        s.F =  tValue;
-        s.F =  tRef;
-        s.F =  tOut;
-        s.F =  tIn;
+        F = ref tValue;
+        F = ref tRef;
+        F = ref tOut;
+        F = ref tIn; // 1
     }
+    object P
+    {
+        init
+        {
+            F = ref GetRef();
+            F = ref GetRefReadonly(); // 2
+        }
+    }
+    static ref T GetRef() => throw null;
+    static ref readonly T GetRefReadonly() => throw null;
 }";
             // PROTOTYPE: Report a ref-safe-to-escape error for: F = ref tValue;
             var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition });
             comp.VerifyEmitDiagnostics(
                 // (10,17): error CS8331: Cannot assign to variable 'in T' because it is a readonly variable
-                //         F = ref tIn;
+                //         F = ref tIn; // 1
                 Diagnostic(ErrorCode.ERR_AssignReadonlyNotField, "tIn").WithArguments("variable", "in T").WithLocation(10, 17),
-                // (20,21): error CS1510: A ref or out value must be an assignable variable
-                //             F = ref GetValue();
-                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "GetValue()").WithLocation(20, 21),
-                // (22,21): error CS8331: Cannot assign to method 'S<T>.GetRefReadonly()' because it is a readonly variable
-                //             F = ref GetRefReadonly();
-                Diagnostic(ErrorCode.ERR_AssignReadonlyNotField, "GetRefReadonly()").WithArguments("method", "S<T>.GetRefReadonly()").WithLocation(22, 21),
-                // (37,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
-                //         s.F =  ref tValue;
-                Diagnostic(ErrorCode.ERR_AssgReadonly, "s.F").WithLocation(37, 9),
-                // (38,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
-                //         s.F =  ref tRef;
-                Diagnostic(ErrorCode.ERR_AssgReadonly, "s.F").WithLocation(38, 9),
-                // (39,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
-                //         s.F =  ref tOut;
-                Diagnostic(ErrorCode.ERR_AssgReadonly, "s.F").WithLocation(39, 9),
-                // (40,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
-                //         s.F =  ref tIn;
-                Diagnostic(ErrorCode.ERR_AssgReadonly, "s.F").WithLocation(40, 9));
+                // (17,21): error CS8331: Cannot assign to method 'S<T>.GetRefReadonly()' because it is a readonly variable
```

Comment:
This diagnostic message confuses the heck out of me every time I see it. It is correct to give an error here, but it would be SO much better if we could say something like... "can't ref-assign a readonly reference to a writable ref variable.".

Reply:
Added a PROTOTYPE comment to consider changing the text.